sdk/servicebus/azure-messaging-servicebus/src/samples/java/com/azure/messaging/servicebus/DeadletterQueueSample.java

```diff
@@ -0,0 +1,222 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.servicebus;
+
+import com.azure.messaging.servicebus.models.ServiceBusReceiveMode;
+import com.nimbusds.oauth2.sdk.ParseException;
+import com.nimbusds.oauth2.sdk.util.JSONObjectUtils;
+import net.minidev.json.JSONObject;
+import org.junit.jupiter.api.Test;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+/**
+ * Sample demonstrates how to dead letter within an Azure Service Bus Queue.
+ */
+public class DeadletterQueueSample {
+    String connectionString = System.getenv("AZURE_SERVICEBUS_NAMESPACE_CONNECTION_STRING");
+    String queueName = System.getenv("AZURE_SERVICEBUS_SAMPLE_QUEUE_NAME");
+
+    /**
+     * Main method to show how to dead letter within an Azure Service Bus Queue.
+     *
+     * @param args Unused arguments to the program.
+     * @throws InterruptedException If the program is unable to sleep while waiting for the receive to complete.
+     */
+    public static void main(String[] args) throws InterruptedException {
+        DeadletterQueueSample sample = new DeadletterQueueSample();
+        sample.run();
+    }
+
+    /**
+     * Run method to invoke this demo on how to dead letter within an Azure Service Bus Queue.
+     *
+     * @throws InterruptedException If the program is unable to sleep while waiting for the receive to complete.
+     */
+    @Test
+    public void run() throws InterruptedException {
+        ServiceBusSenderClient senderClient = new ServiceBusClientBuilder()
+            .connectionString(connectionString)
+            .sender()
+            .queueName(queueName)
+            .buildClient();
+
+        // max delivery-count scenario
+        sendMessagesAsync(senderClient, 1);
+        deadLetterByExceedingMaxDelivery(connectionString, queueName);
+
+        // fix-up scenario
+        sendMessagesAsync(senderClient, Integer.MAX_VALUE);
+        this.receiveAndDeadletterMessagesAsync(connectionString, queueName);
+        this.pickUpAndFixDeadletters(connectionString, queueName, senderClient);
+
+        senderClient.close();
+    }
+
+    /**
+     * Send {@link ServiceBusMessage messages} to an Azure Service Bus Queue.
+     *
+     * @Param senderAsyncClient Service Bus Sender Client
+     * @Param maxMessages Maximum Number Of Messages
+     */
+    void sendMessagesAsync(ServiceBusSenderClient senderAsyncClient, int maxMessages) {
```

Comment:
Should we name the parameter as `senderClient` instead of `senderAsyncClient`? given it uses sync-client, also the sample function has the word async in it.

Reply:
According to your comment, the parameter name is corrected to [senderClient]